controlpanel/api/cluster.py

```diff
@@ -346,3 +356,120 @@ def get_repository(user, repo_name):
         return github.get_repo(repo_name)
     except GithubException.UnknownObjectException:
         return None
+
+
+class ToolDeploymentError(Exception):
+    pass
+
+
+def deploy_tool(tool, user, **kwargs):
+    values = {
+        'username': user.username.lower(),
+        'Username': user.username.lower(),  # XXX backwards compatibility
+        'aws.iamRole': user.iam_role_name,
+        'toolsDomain': settings.TOOLS_DOMAIN,
+    }
+
+    # generate per-deployment secrets
+    for key, value in tool.values.items():
+        if value == '<SECRET_TOKEN>':
+            tool.values[key] = secrets.token_hex(32)
+
+    values.update(tool.values)
+    values.update(kwargs)
+    values = ','.join(f'{key}={val}' for key, val in values.items())
+
+    try:
+        return helm.upgrade_release(
+            f'{tool.chart_name}-{user.slug}',
+            f'mojanalytics/{tool.chart_name}',  # XXX assumes repo name
```

Comment:
I actually now realise that we use this `mojanalytics`/repo name a lot. Maybe worth moving to a config (happy to have that default to `mojanalytics` so we don't have to add config, etc...) or at least a constant?